projects/extension/build.py

```diff
@@ -530,38 +530,34 @@ def install_py() -> None:
     python_install_dir().mkdir(exist_ok=True)
     version = this_version()
     version_target_dir = python_install_dir().joinpath(version)
-    if version_target_dir.exists():
-        # if it already exists, assume the dependencies have already been installed
-        # and just install *our* code. this is for development workflow speed
-        d = version_target_dir.joinpath("ai")  # delete module if exists
-        if d.exists():
-            shutil.rmtree(d)
-        for d in version_target_dir.glob(
-            "pgai-*.dist-info"
-        ):  # delete package info if exists
-            shutil.rmtree(d)
-        bin = "pip3" if shutil.which("uv") is None else "uv pip"
-        cmd = f'{bin} install -v --no-deps --compile --target "{version_target_dir}" "{ext_dir()}"'
-        subprocess.run(
-            cmd,
-            check=True,
-            shell=True,
-            env=os.environ,
-            cwd=str(ext_dir()),
-        )
-    else:
+    bin = "pip3" if shutil.which("uv") is None else "uv pip"
+    if not version_target_dir.exists():
+        # if the target directory doesn't exist, create it and install dependencies
         version_target_dir.mkdir(exist_ok=True)
-        bin = "pip3" if shutil.which("uv") is None else "uv pip"
-        cmd = (
-            f'{bin} install -v --compile --target "{version_target_dir}" "{ext_dir()}"'
-        )
+        cmd = f'{bin} install -r requirements-lock.txt -v --target "{version_target_dir}"',
         subprocess.run(
             cmd,
             check=True,
             shell=True,
             env=os.environ,
             cwd=str(ext_dir()),
         )
+    # install (or re-install) pgai
+    d = version_target_dir.joinpath("ai")  # delete module if exists
+    if d.exists():
+        shutil.rmtree(d)
+    for d in version_target_dir.glob(
+        "pgai-*.dist-info"
+    ):  # delete package info if exists
+        shutil.rmtree(d)
+    cmd = f'{bin} install -v --no-deps --compile --target "{version_target_dir}" "{ext_dir()}"'
+    subprocess.run(
+        cmd,
+        check=True,
+        shell=True,
+        env=os.environ,
+        cwd=str(ext_dir()),
+    )
```

Comment:
Re-iterating my question from the thread:
Do we actually need the code to be a package? Can't we just do `pip install -r requirements-lock.py` and copy the python source files into e.g. a python directory and be done with it?

Reply:
TBH I don't know - what do we gain from that approach vs. this?

Reply:
> Can't we just do pip install -r requirements-lock.py and copy the python source files into e.g. a python directory and be done with it?

If this is what we are gonna ask users to do in case they want to use pgai library in their own code base, I don't think it is viable.

Reply:
No, but it also doesn't make sense to have pinned/locked deps in a lib.

Reply:
That is to say: this approach doesn't impose our locked deps on lib users. We do still have the issue of pretty heavily-pinned deps, which we should perhaps consider relaxing a little for users of the lib.

Reply:
>TBH I don't know - what do we gain from that approach vs. this?

We'd remove a step from the installation, which removes the dependency of setuptools or any other build tool.

And no this has nothing to do with the pgai library. I'm just questioning if the extensions python code should rly be handled like a "library" which is what we are doing right now. Or if it actually is an application.